swing-lib/src/main/java/org/triplea/swing/SwingComponents.java

```diff
@@ -438,4 +440,22 @@ public static JComponent leftBox(final JComponent c) {
     b.add(Box.createHorizontalGlue());
     return b;
   }
+
+  public static void addKeyBinding(
+      final JFrame frame, final KeyStroke keyStroke, final Runnable action) {
+    final JComponent component = (JComponent) frame.getContentPane();
+    addKeyBinding(component, keyStroke, action);
+  }
+
+  public static void addKeyBinding(
+      final JDialog component, final KeyStroke keyStroke, final Runnable action) {
+    addKeyBinding(component.getRootPane(), keyStroke, action);
+  }
+
+  private static void addKeyBinding(
+      final JComponent component, final KeyStroke keyStroke, final Runnable action) {
+    final String keyBindingIdentifier = UUID.randomUUID().toString();
+    component.getInputMap(JComponent.WHEN_IN_FOCUSED_WINDOW).put(keyStroke, keyBindingIdentifier);
+    component.getActionMap().put(keyBindingIdentifier, SwingAction.of(e -> action.run()));
```

Comment:
Any Idea what sets this apart from using a common listener?
Couldn't find a real answer online